Fix for spacing of sitetree panel, reportAdmin, and toggle on Member details page

clarkepaul opened the pull request:
Includes improvements to spacing of buttons in the toolbar.

chillu commented on admin/client/src/styles/legacy/_forms.scss:
```diff
@@ -494,8 +494,8 @@ button.loading {
 	&.ss-ui-button-constructive[class*="font-icon-"] {
-		padding: 5px 8px;
-		margin-bottom: $grid-y*1.5;
+		padding: 6px 8px;
```
Why are we breaking out of the grid constants here?

clarkepaul replied:
I should create a variable for it actually as its a standard measurement. Same with the line-height, there is already a variable for it. Will rectify.

chillu commented on admin/client/src/styles/legacy/_style.scss:
```diff
@@ -1614,21 +1614,28 @@ form.member-profile-form {
+
+    .ss-gridfield-buttonrow {
+      margin-left: -20px;
+      margin-right: -20px;
```
Instead of all of this negative margin malarkey, we should simply remove `.cms-panel-padded` from the container here, right?

chillu commented on admin/client/src/styles/legacy/_style.scss:
```diff
@@ -1613,28 +1613,37 @@ form.member-profile-form {
+    margin-right: -$grid-gutter-width-half;
+  }
+
+  &.cms-panel-padded {
```
Could this be removed if we got rid of `.cms-panel-padded` in ReportAdmin?

clarkepaul replied:
You could but I think the alternative is worse, if you take a look at the _broken links report_ you will see it has no actions toolbar at the top meaning fields are touching the top. We can't add spacing above the :first-child as it sometimes applies it to hidden messages and fields  and it's hard to tell if they are active or not. 

This probably wouldn't be the way I'd do it if I could adjust all of the HTML but at the moment each section of the ReportAdmin & CMS has completely different markup structure (which we will need to standardise) making it hard to maintain.

The Way I'm approaching this only the first toolbar whether it is part of a gridfield will be negatively spaced, everything else will have appropriate spacing above and to the sides. This means no matter what the panel contains it should look okay.

chillu replied:
Talked to Paul about this, see comment on https://github.com/silverstripe/silverstripe-cms/issues/1491#issuecomment-245161859

clarkepaul commented:
Rebased with master to fix failing tests, and added a fix to the upload layout as mentioned here https://github.com/silverstripe/silverstripe-cms/issues/1491#event-794987275

clarkepaul commented:
Added fix for spacing above campaign alert, area was using legacy classes so updated them to the newer panel classes.
After
![image](https://cloud.githubusercontent.com/assets/555033/18659364/39601ff4-7f5e-11e6-8cd6-ac15469aaaf3.png)

flamerohr commented on admin/client/src/styles/legacy/_style.scss:
```diff
@@ -1,24 +1,24 @@
- * Core Styles.
- * ---------------------------------------------------- */
+//
+// This file defines most styles of the CMS: Colors, fonts, backgrounds,
```
How come this changed to `//`?

flamerohr commented on admin/client/src/styles/legacy/_style.scss:
```diff
@@ -1,24 +1,24 @@
+
+// ----------------------------------------------------
+// Core Styles.
+// ---------------------------------------------------- */
```
still has the `*/` at the end ;)

flamerohr commented on admin/client/src/styles/legacy/_style.scss:
```diff
@@ -761,14 +775,14 @@ body.cms {
  * -------------------------------------------- */
```
seems weird that this wasn't changed to `//` as well?
